### PR TITLE
[socket.io] Improve handshake types

### PR DESCRIPTION
**Pull request by ali-furkan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://socket.io/docs/v3/server-socket-instance/#Socket-handshake

### CODE REVIEW

File: types/socket.io/index.d.ts [hunk 1]
```diff
@@ -820,7 +825,7 @@ declare namespace SocketIO {
         /**
          * Any query string parameters in the request url
          */
-        query: any;
+        query: { [props: string]: any };
```
Comment (murbanowicz) [inline types/socket.io/index.d.ts:828]: Please change it to Record to make it more readable.
I would say all query records are `string`

File: types/socket.io/index.d.ts [hunk 2]
```diff
@@ -787,6 +787,11 @@ declare namespace SocketIO {
          */
         headers: any;
 
+        /**
+         * The authentication payload
+         */
+        auth: { [props: string]: any };
```
Comment (murbanowicz) [inline types/socket.io/index.d.ts:793]: Is not the `token` always there? Please try to narrow it as much as possible.

**Review by murbanowicz — CHANGES_REQUESTED** [review]

File: types/socket.io/index.d.ts (hunk 2 above)
Comment (ali-furkqn) [reply types/socket.io/index.d.ts:793]: Oops! I thought it may has different props.

**Review by murbanowicz — APPROVED** [review]
Looks ok to me.

**Comment by ali-furkqn** [comment]
Ready to merge

**Pull request merged**